consensus/qbft/src/reference-test/java/org/hyperledger/besu/consensus/qbt/test/BftExtraDataRlpTest.java

```diff
@@ -0,0 +1,100 @@
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.core.LogsBloomFilter;
+import org.hyperledger.besu.testutil.JsonTestParameters;
+
+import java.util.Collection;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class BftExtraDataRlpTest {
+  private static final String TEST_CONFIG_PATH = "BftExtraDataRLPTests/";
+  private final BftExtraDataTestCaseSpec spec;
+  private static final JsonProvider JSON_PROVIDER = new JsonProvider();
+
+  @Parameterized.Parameters(name = "Name: {0}")
+  public static Collection<Object[]> getTestParametersForConfig() {
+    return JsonTestParameters.create(
+            BftExtraDataTestCaseSpec.class, JSON_PROVIDER.getObjectMapper())
+        .generate(TEST_CONFIG_PATH);
+  }
+
+  public BftExtraDataRlpTest(
+      final String name, final BftExtraDataTestCaseSpec spec, final boolean runTest) {
+    this.spec = spec;
+    assumeTrue("Test [" + name + "] was blacklisted", runTest);
+  }
+
+  @Test
+  public void encode() {
+    final BftExtraData bftExtraData = spec.getBftExtraData();
+    assertThat(bftExtraData.encode()).isEqualTo(spec.getRlp());
+  }
+
+  @Test
+  public void decode() {
+    final BlockHeader blockHeader =
+        setHeaderFieldsExceptForExtraData().extraData(spec.getRlp()).buildBlockHeader();
+    final BftExtraData bftExtraData = BftExtraData.decode(blockHeader);
+    final BftExtraData expectedExtraData = spec.getBftExtraData();
+
+    assertThat(bftExtraData.getVanityData()).isEqualTo(expectedExtraData.getVanityData());
+    assertThat(bftExtraData.getRound()).isEqualTo(expectedExtraData.getRound());
+    assertThat(bftExtraData.getValidators())
+        .containsExactlyInAnyOrder(expectedExtraData.getValidators().toArray(Address[]::new));
+    assertThat(bftExtraData.getVote()).isEqualTo(bftExtraData.getVote());
+    assertThat(bftExtraData.getSeals())
+        .containsExactlyInAnyOrder(
+            expectedExtraData.getSeals().toArray(SECP256K1.Signature[]::new));
+  }
+
+  private static BlockHeaderBuilder setHeaderFieldsExceptForExtraData() {
+    final BlockHeaderBuilder builder = new BlockHeaderBuilder();
+    builder.parentHash(
+        Hash.fromHexString("0xa7762d3307dbf2ae6a1ae1b09cf61c7603722b2379731b6b90409cdb8c8288a0"));
+    builder.ommersHash(
+        Hash.fromHexString("0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347"));
+    builder.coinbase(Address.fromHexString("0x0000000000000000000000000000000000000000"));
+    builder.stateRoot(
+        Hash.fromHexString("0xca07595b82f908822971b7e848398e3395e59ee52565c7ef3603df1a1fa7bc80"));
+    builder.transactionsRoot(
+        Hash.fromHexString("0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"));
+    builder.receiptsRoot(
+        Hash.fromHexString("0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"));
+    builder.logsBloom(
```

Comment:
Are these values just needed to create a valid block header? These values aren't used for test are they?

Reply:
that is correct, only to create a valid blockheader. (so that Blockheader.getExtraData works).